src/handlers/sign_tx.rs

```diff
@@ -77,3 +77,85 @@ pub fn handler(mut stream: SingleTxStream<'_>) -> Result<Signature, AppSW> {
 
     finalize_sign::end(stream, &path)
 }
+
+use ledger_device_sdk::libcall::swap::CreateTxParams;
+
+pub fn handler_swap(
```

Comment:
Add a comment to this function to explain the context
Also I see that it has no factorization with the main signing flow, if that's the case it's a bit unfortunate